### PR TITLE
feat: add client impl, build endpoints from config

### PR DESCRIPTION
**Pull request by heiliuchao**
implement the ping interface and fix 2 bugs based on review comments.

### CODE REVIEW

File: opengemini_client/client_impl.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    def __init__(self, config: Config, endpoints: list[Endpoint]):
+        """
+        OpenGemini client
+        :param config:  config object
+        :param endpoints:  urls
+        """
+        self.config = config
+        self.endpoints = endpoints
+        self.prev_idx = AtomicInt(5)
+        self.data_chan = {}
```
Comment (Chenxulin97) [inline opengemini_client/client_impl.py]: what's the usage of this variant?

File: opengemini_client/models.py [hunk 2]
```diff
@@ -59,3 +53,61 @@ class Query:
...
+    batch_config: BatchConfig
+    timeout: int
+    connection_timeout: int
+    gzip_enabled: bool
+    tls_enabled: bool
+    tls_config: bool
+
+
+@dataclass
+class Endpoint:
```
Comment (hezhangjian) [inline opengemini_client/models.py]: we already have Address, why we still need Endpoint>

Comment (heiliuchao) [reply opengemini_client/models.py]: Reserve it,but remove the attribute of endpoints from the ClientImpl class

File: opengemini_client/client_impl.py (hunk 1 above)
Comment (heiliuchao) [reply opengemini_client/client_impl.py]: accept it. remove prex_idx and data_chan attributes

**Comment by hezhangjian** [comment]
1. The code style is to return a struct and an error, and the definition is also to return a struct and an error, but it uses raise error instead.
2. tls_config should not be a bool type.
3. For timeout, wouldn't it be better to use Python's delta?

**Review by hezhangjian — APPROVED** [review]

**Pull request merged**